sources/packages/backend/libs/services/src/disbursement-overaward/disbursement-overaward.service.ts

```diff
@@ -1,35 +1,55 @@
 import { Injectable } from "@nestjs/common";
-import { DataSource, EntityManager } from "typeorm";
-import { RecordDataModelService, DisbursementOveraward } from "@sims/sims-db";
+import { EntityManager, Repository } from "typeorm";
+import { DisbursementOveraward } from "@sims/sims-db";
 import {
   AwardOverawardBalance,
   StudentOverawardBalance,
 } from "./disbursement-overaward.models";
+import { InjectRepository } from "@nestjs/typeorm";
 
 /**
  * Service layer for Student Application disbursement schedules.
  */
 @Injectable()
-export class DisbursementOverawardService extends RecordDataModelService<DisbursementOveraward> {
-  constructor(dataSource: DataSource) {
-    super(dataSource.getRepository(DisbursementOveraward));
+export class DisbursementOverawardService {
+  constructor(
+    @InjectRepository(DisbursementOveraward)
+    private readonly disbursementOverawardRepo: Repository<DisbursementOveraward>,
+  ) {}
+
+  /**
+   * Checks if the student has any positive overaward balance for any award.
+   * @param studentId student to be checked.
+   * @returns true if the student has any positive overaward balance value
+   * for any award, otherwise, false.
+   */
+  async hasOverawardBalance(studentId: number): Promise<boolean> {
+    const overawards = await this.getOverawardBalance([studentId]);
+    if (!overawards[studentId]) {
+      return false;
+    }
+    return Object.values(overawards[studentId]).some(
```

Comment:
This is an effective balance right. Consider this scenario where student has an overaward and they pay the overaward back.

```
{
CSLF:100  //Re-assessment overaward
CSLF:-100 //Manual entry
}

```
The logic here will return true I believe. We must use reduce here isn't it?

Let me know If I am missing something.

Reply:
The method `getOverawardBalance` returns the SUM for every award. The "reduce", as you mentioned, happens on DB.
The format of the result will be something as below. Please let me know if further clarification is needed.
```ts
/**
 * Total overawards balances per awards indexed per student id.
 * @example considering 123 and 456 as students ids.
 * 123: {
 *    CSLF: 659,
 *    BCSL: 6565,
 * },
 * 456: {
 *    CSLF: 9898,
 * },
 */
```

Reply:
@andrewsignori-aot do you mean that the value doesn't come to JS?

Reply:
Yes, for instance, for Part-Time Canada Loan (CSLF) there will always none to one records and never more than on for the particular student due to the `.groupBy("student.id").addGroupBy("disbursementOveraward.disbursementValueCode")` on `getOverawardBalance`